www/perl-lib/SMT/Rest/Base.pm

```diff
@@ -120,31 +119,7 @@ sub parse_args($)
 sub read_post
 {
     my $self = shift;
-
-    my $bb = APR::Brigade->new($self->request()->pool, $self->request()->connection->bucket_alloc);
-
-    my $data = '';
-    my $seen_eos = 0;
-    do {
-        $self->request()->input_filters->get_brigade($bb, Apache2::Const::MODE_READBYTES,
-                                                     APR::Const::BLOCK_READ, IOBUFSIZE);
-
-        for (my $b = $bb->first; $b; $b = $bb->next($b)) {
-            if ($b->is_eos) {
-                $seen_eos++;
-                last;
-            }
-
-            if ($b->read(my $buf)) { $data .= $buf; }
-            $b->remove; # optimization to reuse memory
-        }
-    } while (!$seen_eos);
-
-    $bb->destroy;
-    # Fake an empty hash in case we get no data
-    $data = '{}' if ! $data;
-    $self->request()->log->info("Got content: $data");
-    return $data;
+    return SMT::Utils::read_post($self->request());
```

Comment:
Here we need to return a string which simulate and empty hash.

> my $data = SMT::Utils::read_post($self->request());
> $data = '{}' if ! $data;
> return $data;

would be better.